### PR TITLE
Implement `compact` config formatters

### LINKED ISSUE #238: Implement "Compact" config formatters
**Description**

Currently, formatters like the `JSON` formatter output a "pretty printed" version of the actual JSON with newlines and indentation. In this feature, we aim to introduce a "compact" version of the same JSON, primarily for performance, parsing, and space considerations. Additionally, we should assess if our other existing formatters also possess a potential "compact" mode.

**Motivation**

Enhancing the flexibility of our formatters suite and catering to diverse use cases where compact representation could be more beneficial due to reduced payload sizes and quicker parse times.

**Additional context**

- https://github.com/configu/configu/blob/main/ts/packages/lib/src/formatters/configFormatters.ts#L72

### PR DESCRIPTION
**Pull request by Sadaf-A**
resolves #238

### CODE REVIEW

File: ts/packages/lib/src/formatters/ConfigFormat.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 export type ConfigFormat =
   | 'JSON'
+  | 'CompactJSON'
```
Comment (rannn505) [inline ts/packages/lib/src/formatters/ConfigFormat.ts:3]: Im still not sure regarding the naming, I will come up with a pool and we well let everyone vote on this 🙏🏽

**Review by RichardAkman — APPROVED** [review]
After internal discussions, we decided to approve this PR as is. We might do some internal fine-tuning later but regardless, great work! Always a pleasure to see your contributions @Sadaf-A 🌟

**Comment by RichardAkman** [comment]
@Sadaf-A This feature will be available once we release the next version of CLI, I assume it will happen in the next few days.

**Comment by Sadaf-A** [comment]
> @Sadaf-A This feature will be available once we release the next version of CLI, I assume it will happen in the next few days. 

awesome!

**Pull request merged**